### PR TITLE
new lint: vec_resize_to_zero

### PR DESCRIPTION
**Pull request by montrivo**
implements #5444

changelog: new lint vec_resize_to_zero

### CODE REVIEW

**Comment by yaahc** [comment]
This looks good btw, just needs to be rebased and have a CI run that doesn't fall over.

cc @flip1995

File: clippy_lints/src/vec_resize_to_zero.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+impl<'a, 'tcx> LateLintPass<'a, 'tcx> for VecResizeToZero {
+    fn check_expr(&mut self, cx: &LateContext<'a, 'tcx>, expr: &'tcx Expr<'_>) {
+        if_chain! {
+            if let hir::ExprKind::MethodCall(_, _, ref args) = expr.kind;
+            if let Some(method_def_id) = cx.tables.type_dependent_def_id(expr.hir_id);
+            if match_def_path(cx, method_def_id, &paths::VEC_RESIZE) && args.len() == 3;
+            if let ExprKind::Lit(Spanned { node: LitKind::Int(0, _), .. }) = args[1].kind;
+            if let ExprKind::Lit(Spanned { node: LitKind::Int(..), .. }) = args[2].kind;
+            then {
+                span_lint(cx, VEC_RESIZE_TO_ZERO, expr.span, "this empties the vector. It could be an argument inversion mistake. If not, call `clear()` instead.");
```
Comment (flip1995) [inline clippy_lints/src/vec_resize_to_zero.rs:39]: I would suggest to split up the message and suggestion:
```suggestion
                span_lint_and_then(
                    cx,
                    VEC_RESIZE_TO_ZERO,
                    expr.span,
                    "empty a vector with `resize`",
                    |db| {
                        db.help("the arguments could be inverted...");
                        db.span_suggestion(
                            resize_span,
                            "...or you can use `clear` to empty the vector",
                            "clear()",
                            Applicability::MaybeIncorrect,
                        );
                    },
                );
```

Note, that you also have to get the span of the method call as `resize_span` for the suggestion to work.

Comment (montrivo) [reply clippy_lints/src/vec_resize_to_zero.rs]: Suggestion added

**Comment by montrivo** [comment]
Should this lint move to the `methods.rs` module ?

**Comment by flip1995** [comment]
I think having this in its own module is fine. The methods module is more about methods (chaining) in general, while this is more about arguments to a specific method.

**Pull request merged**